Library/Homebrew/test/test_uninstall.rb

```diff
@@ -1,8 +1,135 @@
 require "helper/integration_command_test_case"
 
 class IntegrationCommandTestUninstall < IntegrationCommandTestCase
+  def setup
+    super
+    @f1_path = setup_test_formula "testball_f1", <<-CONTENT
+      def install
+        FileUtils.touch prefix/touch("hello")
+      end
+    CONTENT
+    @f2_path = setup_test_formula "testball_f2", <<-CONTENT
+      depends_on "testball_f1"
+
+      def install
+        FileUtils.touch prefix/touch("hello")
+      end
+    CONTENT
+  end
+
+  def f1
+    Formulary.factory(@f1_path)
+  end
+
+  def f2
+    Formulary.factory(@f2_path)
+  end
+
   def test_uninstall
     cmd("install", testball)
     assert_match "Uninstalling testball", cmd("uninstall", "--force", testball)
+    assert_empty Formulary.factory(testball).installed_kegs
+  end
+
+  def test_uninstall_with_unrelated_missing_deps_in_tab
+    setup_test_formula "testball"
+    run_as_not_developer do
+      cmd("install", testball)
```

Comment:
This is failing because `testball` and `"testball"` are not interchangeable in this case. Additionally, I think all the new tests in this file should be made into non-integration tests instead; the integration tests are meant just as an overall smoke test and are too slow to be used for more complex cases like this.